pkg/daemon/winsvc/winsvc.go

```diff
@@ -28,11 +30,27 @@ func EnsureServiceState(service Service, state svc.State) error {
 	}
 	switch state {
 	case svc.Running:
-		return service.Start()
+		err = service.Start()
+		if err != nil {
+			return err
+		}
 	case svc.Stopped:
 		_, err = service.Control(svc.Stop)
-		return err
+		if err != nil {
+			return err
+		}
 	default:
 		return errors.New("unexpected state request")
 	}
+	// wait for the state change to actually take place
+	return wait.PollImmediate(retry.Interval, retry.ResourceChangeTimeout, func() (bool, error) {
+		status, err := service.Query()
+		if err != nil {
+			return false, errors.Wrap(err, "error querying service state")
```

Comment:
Please add the service name here unless you know that the original err will have that info.

Reply:
Getting the name would require another API call to service.Config(). Since this is a function on the service obj itself rather than from the manager, I think it's fine to leave the service name appending to call site error handling.